### PR TITLE
Simple Vehicle Framework

### PR DESCRIPTION
**Pull request by RemieRichards**
## Vehicles:
- Converted janicarts into a simple to use vehicle framework, `/obj/vehicle`
- `/obj/vehicle` has customisable speed
- added an `/obj/vehicle/atv/turret` subtype that has a mounted gun turret on the back
- vehicles are no longer `/obj/structure/bed/chair/janicart` subtypes, we can all sleep a little easier now
## Turret ATV:

turrets

### CODE REVIEW

File: code/modules/vehicles/atv.dm [hunk 1]
```diff
@@ -0,0 +1,81 @@
+
+/obj/vehicle/atv
+	name = "All-terrain vehicle"
```
Comment (MrPerson) [inline code/modules/vehicles/atv.dm]: Name should be lowercase

**Comment by Man-Shroom-** [comment]
Remie why are there two sunnyside eggs on the front of the vehicle where the lights should be

File: code/modules/vehicles/atv.dm (hunk 1 above)
Comment (RemieRichards) [inline code/modules/vehicles/atv.dm]: took from the original (granted I moved it's ```callme``` to it's name as I deprecated that var, but yeah I'll sort it)

**Comment by KazeEspada** [comment]
What about multiperson vehicles and vehicles you get completely inside.

**Comment by RemieRichards** [comment]
@Man-Shroom- no idea, ask @MMMiracles it's his ATV vehicle just with a new backend code + mounted syndieturrets

File: code/modules/vehicles/vehicle.dm [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+
+
+/obj/structure/bed/chair/janicart/secway
+	parent_type = /obj/vehicle/secway
+
+/obj/structure/bed/chair/janicart
+	parent_type = /obj/vehicle/janicart
+
+/obj/structure/bed/chair/janicart/atv
+	parent_type = /obj/vehicle/atv
```
Comment (MrStonedOne) [inline code/modules/vehicles/vehicle.dm]: Commenting to highlight this awesome idea.

File: code/modules/vehicles/pimpin_ride.dm [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+/obj/vehicle/janicart/Move(a, b, flag)
+	..()
+	if(floorbuffer)
+		var/turf/tile = loc
+		if(isturf(tile))
+			tile.clean_blood()
+			for(var/A in tile)
+				if(istype(A, /obj/effect))
+					if(is_cleanable(A))
+						qdel(A)
```
Comment (MrPerson) [inline code/modules/vehicles/pimpin_ride.dm]: I made the Moved() proc just to handle this kind of shit. And also to kill those terrible arg names which are 100% incorrect.

**Comment by Man-Shroom-** [comment]
@KazeEspada I assume the second one would not be too hard since it would just be a normal vehcle but you wouldn't see your own sprite

And for the second one someone could dig up clown cars or make some hacky bullshit like pAI or something where the mob is tied to the object temporarily

File: code/modules/vehicles/pimpin_ride.dm (hunk 3 above)
Comment (RemieRichards) [inline code/modules/vehicles/pimpin_ride.dm]: Oh yeah jesus I didn't even clean that bit up, give me a mo

**Comment by RemieRichards** [comment]
if you want multi-user, I can probably do that later on, just make some form of list of people and one ref to a main "driver"

File: code/modules/vehicles/vehicle.dm (hunk 2 above)
Comment (RemieRichards) [inline code/modules/vehicles/vehicle.dm]: YOUR idea lol

**Comment by AlexCrimson** [comment]
One step closer to replacing MuleBots with a Cargo Buggy.

**Comment by RemieRichards** [comment]
@AlexCrimson with turrets, a GUNCARGO buggy

**Comment by Bawhoppen** [comment]
add the engi-pod, theres already sprites in the code

**Comment by RemieRichards** [comment]
@Bawhoppen later on, because right now I'd either:
* have to hackily copy the breathing code of mechs, which is not happening
* ignore breathing altogether, which is a bit too powerful

after this PR I can work on proper breathing and "inside" vehicles.

**Comment by MMMiracles** [comment]
@Man-Shroom- I-I thought the sprites looked ok..

**Comment by RemieRichards** [comment]
@MMMiracles they look good enough.

File: code/modules/vehicles/vehicle.dm (hunk 2 above)
Comment (optimumtact) [inline code/modules/vehicles/vehicle.dm]: no this is bad

File: code/modules/vehicles/vehicle.dm [hunk 4]
```diff
@@ -0,0 +1,138 @@
+
+/obj/vehicle
+	name = "vehicle"
+	desc = "A basic vehicle, vroom"
+	icon = 'icons/obj/vehicles.dmi'
+	icon_state = "fuckyou"
```
Comment (duncathan) [inline code/modules/vehicles/vehicle.dm]: Lol

Comment (RemieRichards) [inline code/modules/vehicles/vehicle.dm]: I... I forgot I left that it.
Oh well~

Comment (ike709) [inline code/modules/vehicles/vehicle.dm]: Leave it in, please

Comment (RemieRichards) [inline code/modules/vehicles/vehicle.dm]: oh, it's staying :P

Comment (Bawhoppen) [inline code/modules/vehicles/vehicle.dm]: it is necessary honestly

Comment (RemieRichards) [inline code/modules/vehicles/vehicle.dm]: well it serves it's purpose, it makes /obj/vehicle invisible by default, which further explains that you shouldn't create instances of the base class, allowing admins/coders alike to see that it is an abstract class.

so yes it does actually do it's job, but its fun to have it say fuckyou

Comment (Bawhoppen) [inline code/modules/vehicles/vehicle.dm]: I meant it was necessary for coder morale

Comment (RemieRichards) [inline code/modules/vehicles/vehicle.dm]: that too.

File: code/modules/vehicles/vehicle.dm (hunk 2 above)
Comment (tkdrg) [inline code/modules/vehicles/vehicle.dm]: Suggestion: create a new folder/file for this. `legacy_compat.dm` or something. Then we can deprecate those paths and give mappers a deadline to go fix them.

Comment (RemieRichards) [inline code/modules/vehicles/vehicle.dm]: That was the intended goal for this (although it was a folder called __TYPEFUCKERY) if the idea went well (that meant after this PR, but I can do it now)

**Pull request merged**